ssl/t1_enc.c

```diff
@@ -316,6 +324,76 @@ int tls1_change_cipher_state(SSL *s, int which)
         goto err;
     }
 
+#ifndef OPENSSL_NO_KTLS
+    if (s->compress || SSL_TREAT_AS_TLS13(s))
+        goto skip_ktls;
+
+    if (!(which & SSL3_CC_WRITE)) {
+#ifdef SSL_DEBUG
+        printf("\nSkipping ktls for non-write context\n");
+#endif /* SSL_DEBUG */
+        goto skip_ktls;
+    }
+
+    /* ktls supports only the maximum fragment size */
+    if (ssl_get_max_send_fragment(s) != SSL3_RT_MAX_PLAIN_LENGTH)
+        goto skip_ktls;
+
+    /* check that cipher is AES_GCM_128 */
+    if (EVP_CIPHER_mode(c) != EVP_CIPH_GCM_MODE
+        || EVP_CIPHER_key_length(c) != TLS_CIPHER_AES_GCM_128_KEY_SIZE) {
```

Comment:
You can't identify AES_GCM_128 by check EVP_CIPH_GCM_MODE and key size. I mean GCM mode is not exclusive to AES.

Reply:
Done.